src/services/dashboard.js

```diff
@@ -209,86 +209,50 @@ export async function queryIterationSamples(params) {
   });
 }
 
-export async function queryTimeseriesData(params) {
-  const { datastoreConfig, selectedIndices, selectedResults } = params;
+export async function queryTimeseriesData(payload) {
+  const { datastoreConfig, selectedIndices, selectedIterations } = payload;
 
   const endpoint = `${datastoreConfig.elasticsearch}/${parseMonths(
     datastoreConfig,
     datastoreConfig.result_index,
     selectedIndices
   )}/_search?scroll=1m`;
 
-  const iterationSampleRequests = [];
-  selectedResults.forEach(run => {
-    iterationSampleRequests.push(
-      request.post(endpoint, {
-        data: {
-          query: {
-            filtered: {
-              query: {
-                multi_match: {
-                  query: run.id,
-                  fields: ['run.id'],
+  const timeseriesRequests = [];
+  Object.entries(selectedIterations).forEach(([runId, run]) => {
+    Object.entries(run.iterations).forEach(([, iteration]) => {
+      timeseriesRequests.push(
+        request.post(endpoint, {
+          data: {
+            query: {
+              filtered: {
+                query: {
+                  query_string: {
+                    query: `_type:pbench-result-data AND run.id:${runId} AND iteration.number:${
+                      iteration.number
+                    } AND sample.@idx:${iteration.closest_sample -
```

Comment:
It seems we should add `AND sample.measurement_title:${sample.measurement_title} `.

```
[iternation.name]
    [sample.measurement_type]
        [sample.measurement_title]
            [sample.measurement_idx]
                [sample.@idx]
```